### PR TITLE
Created ScriptExtension interface #318

### LINKED ISSUE #318: [FEATURE] Create ScriptExtension interface
### Is your feature request related to a problem?

The `getScriptEngine()` and `getContexts()` extension points communicate an extension's script capabilities to OpenSearch.

### What solution would you like?

Add a new interface `ScriptExtension` with the above two methods, with default implementations matching `ScriptPlugin.java`.

### PR DESCRIPTION
**Pull request by petiveriaalliacea**
### Description
Added a new interface ScriptExtension with the above two methods, with default implementations matching ScriptPlugin.java.

### Issues Resolved
#318 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dbwiddis — CHANGES_REQUESTED** [review]
Thanks for your contribution!

Two minor nits (Spotless check is failing) and this should be GTG.

File: src/main/java/org/opensearch/sdk/ScriptExtension.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * An additional extension point for {@link Extension}s that extends OpenSearch's scripting functionality.
+ *
+ * @opensearch.api
+ */
+public interface ScriptExtension {
```
Comment (saratvemulapalli) [inline src/main/java/org/opensearch/sdk/ScriptExtension.java:26]: @dbwiddis as we move over more extension points, we should probably group all these interfaces in one directory.

**Review by saratvemulapalli — APPROVED** [review]
Thank you for the contribution!!
LGTM other than comments from @dbwiddis

File: src/main/java/org/opensearch/sdk/ScriptExtension.java (hunk 1 above)
Comment (dbwiddis) [reply src/main/java/org/opensearch/sdk/ScriptExtension.java:26]: @saratvemulapalli already would have done it if I could [think of a good package name](https://martinfowler.com/bliki/TwoHardThings.html).

**Review by dbwiddis — CHANGES_REQUESTED** [review]
Gradle check is failing:

> /home/runner/work/opensearch-sdk-java/opensearch-sdk-java/src/main/java/org/opensearch/sdk/ScriptExtension.java:24: error: unknown tag: opensearch.api

We don't (yet) use that annotation on the SDK, see discussion at #168.

So for now just remove this.

**Review by dbwiddis — APPROVED** [review]

**Review by joshpalis — APPROVED** [review]

**Review by saratvemulapalli — APPROVED** [review]

**Pull request merged**